### PR TITLE
1906dev frontend wb

### PR DESCRIPTION
**Pull request by Reiterpallasch**
Trainer views only their associates

### CODE REVIEW

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts [hunk 1]
```diff
@@ -72,6 +79,34 @@ export class AssociateListPageComponent implements OnInit, OnDestroy, AfterViewI
      */
 
     // Add option for none
+    const thingy2 = JSON.parse(this.lsHelp.localStorageItem("currentUser"));
```
Comment (mjrerle) [inline NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts]: refactor

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts [hunk 2]
```diff
@@ -57,6 +58,12 @@ export class AssociateListPageComponent implements OnInit, OnDestroy, AfterViewI
   constructor(private clientService: ClientService, public associateService: AssociateService, public dialog: MatDialog, public lsHelp: Helpers) { }
 
   ngOnInit() {
+   this.lsHelp.removeStorageItem("clientGetAll");
```
Comment (mjrerle) [inline NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts:62]: not clear why this is being done

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts [hunk 3]
```diff
@@ -72,6 +79,34 @@ export class AssociateListPageComponent implements OnInit, OnDestroy, AfterViewI
      */
 
     // Add option for none
+    const thingy2 = JSON.parse(this.lsHelp.localStorageItem("currentUser"));
+    if (thingy2.role === 2){
+      this.associateStatuses.push("");
```
Comment (mjrerle) [inline NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts:85]: why do this

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts [hunk 4]
```diff
@@ -72,6 +79,34 @@ export class AssociateListPageComponent implements OnInit, OnDestroy, AfterViewI
      */
 
     // Add option for none
+    const thingy2 = JSON.parse(this.lsHelp.localStorageItem("currentUser"));
+    if (thingy2.role === 2){
```
Comment (mjrerle) [inline NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts]: i dont understand what is going on in this block of code

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts [hunk 5]
```diff
@@ -123,8 +142,13 @@ export class AssociateListPageComponent implements OnInit, OnDestroy, AfterViewI
 
   ngOnDestroy(): void {
     //Called once, before the instance is destroyed.
+   this.lsHelp.removeStorageItem("checked");
```
Comment (mjrerle) [inline NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts:146]: why do this?

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts [hunk 6]
```diff
@@ -183,9 +207,16 @@ export class AssociateListPageComponent implements OnInit, OnDestroy, AfterViewI
   }
 
   getNextPage() {
-    if (!this.isFetching) {
+    const thingy3 = JSON.parse(this.lsHelp.localStorageItem("currentUser"));
```
Comment (mjrerle) [inline NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts]: refactor

File: NGTrackForce/src/app/components/client-list/client-list.component.ts [hunk 7]
```diff
@@ -74,6 +74,7 @@ export class ClientListComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit() {
+    this.lsHelp.removeStorageItem("clientGetAll");
```
Comment (mjrerle) [inline NGTrackForce/src/app/components/client-list/client-list.component.ts:77]: why do this?

Comment (Reiterpallasch) [reply NGTrackForce/src/app/components/client-list/client-list.component.ts:77]: local window storage needs global helper

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts (hunk 5 above)
Comment (Reiterpallasch) [reply NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts:146]: local window storage needs global helper

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts (hunk 4 above)
Comment (Reiterpallasch) [reply NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts]: checks if user is a trainer and if so, use a different associate service, refactored

File: NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts (hunk 2 above)
Comment (Reiterpallasch) [reply NGTrackForce/src/app/components/associate-list-page/associate-list-page.component.ts:62]: local window storage needs global helper, this is done to fix permanent loading issue

**Review by mjrerle — APPROVED** [review]

**Pull request merged**